dnsutil/util.go

```diff
@@ -0,0 +1,91 @@
+package dnsutil
+
+import "strings"
+
+// AddDomain adds origin to s if s is not already a FQDN.
+func AddDomain(origin, s string) string {
```

Comment:
To be in sync with the names of the functions in `labels.go`, this is better named `AddDomainName` or maybe `AddOrigin`  and then reverse the order of the parameters.